**Autor/a:** Sergio Rodríguez

In [1]:
import pandas as pd
import numpy as np

try:
    df = pd.read_csv("../Data/titanic.csv")
except FileNotFoundError:
    df = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv")

df.shape

(891, 15)

In [2]:
df["age"] = df["age"].fillna(df["age"].median())
df["embarked"] = df["embarked"].fillna(df["embarked"].mode()[0])
df["embark_town"] = df["embark_town"].fillna(df["embark_town"].mode()[0])
df = df.drop(columns="deck")

In [3]:
df["tam_familia"] = df["sibsp"] + df["parch"] + 1

fare_min = df["fare"].min()
fare_max = df["fare"].max()
df["tarifa_norm"] = (df["fare"] - fare_min) / (fare_max - fare_min)

df[["tam_familia", "tarifa_norm"]].describe()

,tam_familia,tarifa_norm
count,891.000000,891.000000
mean,1.904602,0.062858
std,1.613459,0.096995
min,1.000000,0.000000
25%,1.000000,0.015440
50%,1.000000,0.028213
75%,2.000000,0.060508
max,11.000000,1.000000


In [4]:
assert np.isclose(df["tarifa_norm"].min(), 0)
assert np.isclose(df["tarifa_norm"].max(), 1)

print(df["tam_familia"].value_counts().sort_index())
print(df[["tarifa_norm", "tam_familia"]].corr().round(3))
print(df.groupby("pclass")["tarifa_norm"].mean().round(3))

tam_familia
1     537
2     161
3     102
4      29
5      15
6      22
7      12
8       6
11      7
Name: count, dtype: int64
             tarifa_norm  tam_familia
tarifa_norm        1.000        0.217
tam_familia        0.217        1.000
pclass
1    0.164
2    0.040
3    0.027
Name: tarifa_norm, dtype: float64


## 4. Hallazgo

El 60% de los pasajeros (537 de 891) viajaba solo. La tarifa media en 1ª 
clase (0,164) fue casi 6 veces la de 3ª clase (0,027), y la relación entre 
tamaño de familia y tarifa pagada es débil (correlación de 0,217).